src/h5web/toolbar/controls/DomainSlider/DomainSlider.tsx

```diff
@@ -36,21 +33,32 @@ interface Props {
 function DomainSlider(props: Props): ReactElement {
   const { dataDomain, customDomain, disabled, onCustomDomainChange } = props;
 
-  const [appliedDomain, setAppliedDomain] = useAppliedDomain(
-    dataDomain,
-    customDomain
-  );
-
-  const [tooltipOpen, toggleTooltip] = useToggle(false);
-  useKey('Escape', () => toggleTooltip(false));
+  const appliedDomain = useAppliedDomain(dataDomain, customDomain);
 
-  const [extendedMin, extendedMax] = extendDomain(dataDomain, EXTEND_FACTOR);
-  const step = Math.max((extendedMax - extendedMin) / 100, 10 ** -NB_DECIMALS);
+  const sliderScale = useSliderScale(dataDomain);
+  const [sliderDomain, setSliderDomain] = useSliderDomain(appliedDomain);
 
   const isAutoMin = customDomain[0] === undefined;
   const isAutoMax = customDomain[1] === undefined;
   const isAutoscaling = isAutoMin || isAutoMax;
 
+  const [tooltipOpen, toggleTooltip] = useToggle(false);
+  useKey('Escape', () => toggleTooltip(false));
+
+  const handleSliderChange = useCallback(
+    ([min, max]) => {
+      setSliderDomain([sliderScale.invert(min), sliderScale.invert(max)]);
+    },
+    [sliderScale, setSliderDomain]
+  );
+
+  const handleSliderAfterChange = useCallback(
+    ([min, max]) => {
+      onCustomDomainChange([sliderScale.invert(min), sliderScale.invert(max)]);
+    },
+    [sliderScale, onCustomDomainChange]
+  );
```

Comment:
`handleSliderAfterChange` will diverge more from `handleSliderChange` in my next PR, which is why I didn't try to abstract them.